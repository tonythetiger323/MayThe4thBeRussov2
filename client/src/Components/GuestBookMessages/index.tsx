import React from 'react'
import Card from '@material-ui/core/Card'
import CardContent from '@material-ui/core/CardContent'
import Typography from '@material-ui/core/Typography'
import API from '../../utils/API'

interface Message {
  id: string;
  message: string;
  name: string;
}

interface GuestBookMessagesState {
  messages: Message[]
}

class GuestBookMessages extends React.Component<any, GuestBookMessagesState> {
  constructor(props: any) {
    super(props)
      
      this.state = {
        messages: []
      }
  }

  async componentDidMount(){
    const res = await API.getMessages();
    console.log(res.data);
    this.setState({ messages: res.data })
  }

  render(){
    const messages = this.state.messages
    console.log(messages);
    return(
      <div className='messages'>
        { messages && messages.map(toCard) }
      </div>
    )
  }
}

const toCard = (message: Message) => {
  return(
    <div key={message.id} className='message'>
      <Card style={{ width: 300, opacity: 0.6 }}>
        <CardContent>
          <Typography style={{ fontFamily: 'Oswald' }} variant='h6' align='center'>{message.message}</Typography>
          <Typography style={{ fontFamily: 'Oswald' }} variant='body1' align='center'>{message.name} </Typography>
        </CardContent>
      </Card>
      <br/>
    </div>
  )
}

export default GuestBookMessages