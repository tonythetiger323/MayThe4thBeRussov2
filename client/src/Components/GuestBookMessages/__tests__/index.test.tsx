import * as React from 'react'
import { shallow } from 'enzyme'
import GuestBookMessages from '../index'

describe('Guest Book Messages', () => {
  let wrapper: any
  beforeEach(() => (wrapper = shallow(<GuestBookMessages />)))

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})