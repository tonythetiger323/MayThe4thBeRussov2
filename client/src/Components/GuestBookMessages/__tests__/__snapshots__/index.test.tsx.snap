// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Guest Book Messages should render correctly 1`] = `ShallowWrapper {}`;
